File: src/main/java/im/joker/helper/GlobalStateHolder.java
package im.joker.helper;

import im.joker.device.DeviceManager;
import im.joker.session.AuthManager;
import im.joker.store.ReactiveMongodbStore;
import lombok.Data;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.ReactiveStringRedisTemplate;
import org.springframework.stereotype.Component;

@Component
@Data
public class GlobalStateHolder {

    @Autowired
    private AuthManager authManager;
    @Autowired
    private DeviceManager deviceManager;
    @Autowired
    private ReactiveStringRedisTemplate redisTemplate;
    @Autowired
    private ReactiveMongodbStore mongodbStore;
    @Autowired
    private RoomSubscribeManager roomSubscribeManager;
    @Autowired
    private LongPollingHelper longPollingHelper;
    @Autowired
    private EventSyncQueueManager eventSyncQueueManager;

}

File: src/main/kotlin/im/joker/helper/GlobalStateHolder.kt
package im.joker.helper

import im.joker.device.DeviceManager
import im.joker.repository.MongoStore
import org.redisson.api.RedissonReactiveClient
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

/**
 * @Author: mkCen
 * @Date: 2020/11/21
 * @Time: 18:01
 * @Desc:
 */
@Component
class GlobalStateHolder {
    @Autowired
    lateinit var deviceManager: DeviceManager

    @Autowired
    lateinit var redissonClient: RedissonReactiveClient

    @Autowired
    lateinit var mongoStore: MongoStore

    @Autowired
    lateinit var roomSubscribeManager: RoomSubscribeManager

    @Autowired
    lateinit var eventAuthorizationValidator: EventAuthorizationValidator

    @Autowired
    lateinit var syncEventManager: SyncEventManager

    @Autowired
    lateinit var imCache: ImCache

    @Autowired
    lateinit var longPollingHelper: LongPollingHelper


}